### DISCUSSION TITLE
'cisco_xe' device_type not mapping through to ntc-templates properly

### DISCUSSION THREAD

**Discussion by r1chardnoggin** [ktbyers/netmiko#2112] [Q&A] [answered]
Hello,
I'm new here. I'm a network engineer by trade and I'm trying to get into network automation but I'm a noob on that front, so I'm trying to develop my skillset. I have installed netmiko and textfsm. However when I call on textfsm (use_textfsm=True) in my code it fails to use it to parse the output and I just get back unstructured data. I'm not entirely sure what I'm missing. I believe I have installed everything correctly but like I said I'm knew to this arena. Any help would be greatly appreciated.

**Comment by ktbyers** [comment]
Are you using ntc-templates? Does the given command you are trying to parse exist in the index file of ntc-templates?

Assuming you are using ntc-templates, this is probably more of a question that should be directed there.


Regards, Kirk

**Reply by r1chardnoggin** [reply]
I have the ntc-templates/templates directory as well as my export NET_TEXTFSM pointed to that location. I am executing 'show ip interface brief' and that is indeed in the ntc-templates.
Forgive my ignorance, but where do I need to address this question?

**Comment by ktbyers** [comment]
I would `pip install ntc-templates` and make sure you have Netmiko 3.3.2 installed also `pip install netmiko==3.3.2`

Does that fix the issue?

**Reply by r1chardnoggin** [reply]
`pip3 install ntc-templates` shows requirements already satisfied.
`pip3 list | grep netmiko` shows 3.3.2

**Comment by ktbyers** [comment]
Okay, next most likely problem is that the existing TextFSM template doesn't properly handle your particular output. That would be an ntc-template issue:

https://github.com/networktocode/ntc-templates

**Comment by r1chardnoggin** [comment, marked as answer]
I received some assistance over at github.com/networktocode/ntc-templates. I was asked to run the following...
```
output = net_connect.send_command("show ip interface brief")
from ntc_templates.parse import parse_output
output_parsed = parse_output(platform="cisco_ios", command="show ip interface brief", data=output)
print(output_parsed)
```
That produced the desired output. Then I was referred back here. Any thoughts on what I'm missing / doing wrong?

**Comment by r1chardnoggin** [comment]
I think I figured out my problem. I had set `'device_type': 'cisco_xe'` which outputs the correct ouput from the issued command just in unstructured format. When I changed to `'device_type': 'cisco_ios'` textfsm parses correctly and I then receive that data in a structured format. Rookie mistake I guess.

**Comment by ktbyers** [comment]
No, that is something I should fix on my side or at least warn you about. Let me flag it as a bug to see if we can improve that behavior.
